### PR TITLE
tests: interrupt: Keeping IRQ priorities low

### PR DESCRIPTION
**Pull request by spoorthik**
Keeping IRQ0 priority as 1 and IRQ1 priority as 0
so that system timer which of priority 0 in ARC
will be interrupted by IRQ1 of same priority.
In ARM, system timer is of priority 1, hence
making ISR0 priority as 2 and ISR1 priority as 1.
Thus system timer will always be interrupted by
ISR1 in both the architectures.

Fixes: #12147 

Signed-off-by: Spoorthi K <spoorthi.k@intel.com>

### CODE REVIEW

**Comment by spoorthik** [comment]
recheck

**Review by ramakrishnapallala — APPROVED** [review]

File: tests/kernel/interrupt/src/nested_irq.c [hunk 1]
```diff
@@ -5,6 +5,21 @@ struct k_timer timer;
...
+ * In NRF5, RTC system timer is of priority 1 and
+ * in all other architectures, system timer is considered
+ * to be in priority 0.
+ */
+#if defined(CONFIG_ARM)
+	#define ISR0_PRIO 2
+	#define ISR1_PRIO 1
+#else
+	#define ISR0_PRIO 1
+	#define ISR1_PRIO 0
```
Comment (AdithyaBaglody) [inline tests/kernel/interrupt/src/nested_irq.c]: The limit in ARC is 16 priority levels, if I am not mistaken.
And 0 is taken up by the timer. 
The expectation here is to have lower priority than timer.

**Review by AdithyaBaglody — CHANGES_REQUESTED** [review]
Please update the commit and make it a bit more verbose.

File: tests/kernel/interrupt/src/nested_irq.c (hunk 1 above)
Comment (andyross) [reply tests/kernel/interrupt/src/nested_irq.c]: Is the timer defaulting to the **lowest** priority on ARC?  That seems wrong, designwise.  Usually you want timekeeping to happen at high priorities, with a few available above it for devices with latency requirements.

Also x86_64 (to be merged any moment) uses "raw" APIC priority values, meaning that a user interrupt needs to have a priority in the range 2-15 (the first two "priority classes" are are for CPU exceptions).  x86 does the adjustment for you and allows the use of priorities 0-13.

Comment (andyross) [reply tests/kernel/interrupt/src/nested_irq.c]: Also also, xtensa has fixed priorities for each interrupt and ignores the priority argument entirely.  It won't work with this test unless you also select specific IRQs (though it looks like it doesn't run this subtest anyway, as there's no trigger_irq() implementation for it).

**Review by andyross — APPROVED** [review]
My comments notwithstanding, neither xtensa nor x86 (or x86_64 of course) actually hit this subtest.  So really as long as ARM, and ARC are handled correctly (I'll defer to @AdithyaBaglody as to whether the collision on ARC is fixed, I don't know that arch), this patch looks fine to me.

I'm going to separately add an issue to track the fact that this test needs to be enhanced.  There's no reason we can't run it on most/all platforms, and in fact "trigger_irq()" should probably be an API we expose.

**Comment by spoorthik** [comment]
@AdithyaBaglody @andyross : Thanks for the review. I have updated commit message.

Also @AdithyaBaglody, the interrupt priority level range is 0 - 16 as you mentioned. But the CONFIG_NUM_IRQ_PRIO_LEVELS is set to 2. 

`build/zephyr/.config:4:CONFIG_NUM_IRQ_PRIO_LEVELS=2`

**Review by AdithyaBaglody — APPROVED** [review]
LGTM

**Pull request merged**